internal/store/store.go

```diff
@@ -443,8 +444,20 @@ func (s *ObjectStore) referencedList(ctx context.Context, list model.ObjectList,
 		if err != nil {
 			return err
 		}
-		err = s.readByTypeID(ctx, s.store, typ, key[keyPrefixLen:], object)
-		if err != nil {
+
+		// When `opt.ReferenceReverseLookup` is not true, we're extracting <id> from the below key:
+		// `c/.../ix/f/<typ>/<opt.ReferenceID>/<opt.ReferenceType>/<id>`
+		//
+		// When it is true, we're extracting the <id> from the below key:
+		// `c/.../ix/f/<typ>/<id>/<opt.ReferenceType>/<opt.ReferenceID>`
+		id := key[keyPrefixLen:]
+		if opt.ReferenceReverseLookup {
+			startIdx := strings.Index(keyPrefix, persistence.WildcardOperator)
+			endIdx := strings.Index(key[startIdx:], "/")
+			id = key[startIdx : startIdx+endIdx]
```

Comment:
seems a bit fragile.  i think `strings.Cut` has better behaviour when the separator isn't found.

```suggestion
			startIdx := strings.Index(keyPrefix, persistence.WildcardOperator)
			if startIdx < 0 {
				return fmt.Errorf(.....)
			}
			id, _, _ = strings.Cut(key[startIdx:], "/")
```

Reply:
Fair point, updated 😀